pwiz_tools/Skyline/SkylineGraphs.cs

```diff
@@ -1132,7 +1132,16 @@ void GraphSpectrum.IStateProvider.BuildSpectrumMenu(bool isProteomic, ZedGraphCo
 
             // Insert skyline specific menus
             var set = Settings.Default;
-            var control = menuStrip.SourceControl.Parent.Parent as IMzScalePlot;
+
+            Func<Control, IMzScalePlot> traverseParents = null;
```

Comment:
Seems like this could be done with a loop. Using a function pointer and even starting with = null and then assigning it seems odd. Best not to leave other developers scratching their heads wondering, "Why did they do that?"

Reply:
I don't see that you are traversing a tree. Rather you are looping upward through a straight line of parents. Though they are involved in a tree. This is essentially traversing a linked list for your purposes.

Have a look at FormUtil.FindTopLevelOwner() or SequenceTree.GetNodeOfType(). Clearly, Nick and I prefer loops for parent pointer traversal. I don't really see a benefit to using recursion, and I am certain this code could be simpler without it.